frontend/src/routes/Login.tsx

```diff
@@ -50,37 +50,46 @@ const Login: React.FC<Props> = ({ queryRef }) => {
     useNoindexTag();
     const { t } = useTranslation();
     const { currentUser } = usePreloadedQuery(query, queryRef);
+    const router = useRouter();
+    const isLoggedIn = currentUser !== null;
 
-    return <Outer>
-        <div css={{
-            height: "calc(1.7 * var(--header-height))",
-            maxHeight: "35vh",
-            padding: `${HEADER_BASE_PADDING}px 0`,
-            textAlign: "center",
-        }}><Logo /></div>
+    React.useEffect(() => {
+        if (isLoggedIn) {
+            // This isn't working - 
+            // window.sessionStorage.getItem(REDIRECT_STORAGE_KEY) is always null.
```

Comment:
Yes and no. We can make it so that there are useful situation where this is set. 

Please also change `UserBox` around line 79 to also set the session storage item if a login link is given. That means that for example: an admin configures `loginLink` to be sth like `/Shibboleth.sso/Login` or so, i.e. an external login link. Once that login is done, the admin can just tell Shibboleth to forward the logged in user to `/~login`. And then your code would make it so that the user is redirected to the correct page.

And then you can remove this very comment.

Reply:
I couldn't quite figure out how to do this unfortunately. I would somehow need to check if there is a `loginLink`(I guess by checking for the existence of `CONFIG.auth.loginLink`) and then call `window.sessionStorage.setItem(REDIRECT_STORAGE_KEY, window.location.href)`. I'm pretty sure you don't mean this should be done inside the `onClick` handler (as there it checks the opposite condition and then calls the function, and also because I don't think the handler would be called anyway as there wouldn't be anything being clicked). So there would need to be some handler which checks sth like a route change? But I don't think that exists. Maybe you can give me another hint or suggestion?

Reply:
I guess what @LukasKalbertodt meant is that you just **remove** the condition in the `onClick` handler, so that it stores the current URL in session storage regardless of whether an external login link is configured or not.

You are right that this doesn't change anything for the cases where the user doesn't press the button (copying a link, bookmarks, etc.) **but** system admins using an external login link can now redirect users back to just `/~login` and have the user end up in the right place after clicking on the button. Right now this only works when no external login is configured.

Reply:
Ahh alright. Thats's what I was figuring at first, but then started overthinking it a little too much I guess. Thank you for the clarification.